### PR TITLE
[Blazor] Globalization & localization - correct async JS interop

### LINKED ISSUE #32365: CultureSelector should call JS interop asynchronously
### Description

Although the first demo is "client-side only" (WASM), I believe the `CultureSelector.razor` versions in the article (definitely at least the BWA version) should use JS interop asynchronously. This means JavaScript should not be called from the `set_Culture` property setter, but from an asynchronous event handler, such as `@bind-Culture:after` or `bind-Culture:set`.

For consistency, all `CultureSelector` components should be refactored to remove any logic from `set_Culture` and instead utilize a separate event handler.

If you agree, I will prepare a PR for this.

cc @guardrex

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/globalization-localization.md

### Document ID

d6f07538-228e-9f96-680f-6c324caf11d6

### Article author

@guardrex

### PR DESCRIPTION
**Pull request by hakenr**
fixes #32365

cc @guardrex 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/globalization-localization.md](https://github.com/dotnet/AspNetCore.Docs/blob/1f2d640b10ff19558dc458122ad5eb497c3a203e/aspnetcore/blazor/globalization-localization.md) | [ASP.NET Core Blazor globalization and localization](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/globalization-localization?branch=pr-en-us-32433) |

### CODE REVIEW

**Comment by hakenr** [comment]
@guardrex There is still one more instance of the `Culture` property in the _Dynamically set the server-side culture by user preference_ section where the logic resides in the property's getter and setter (a practice neither I nor the documentation and code analyzers recommend). However, the setter does not invoke async JS interop.

Please let me know if I should also update this demo to use `@bind:after` (either in this PR or a separate one).

**Comment by guardrex** [comment]
> Please let me know if I should also update this demo to use `@bind:after` ...

Yes, please feel free to address that on this PR. Fortunately, the version flipping in and out of 8.0 won't affect your update 😅.

**Comment by hakenr** [comment]
Ready for review

File: aspnetcore/blazor/globalization-localization.md [hunk 1]
```diff
@@ -702,32 +760,86 @@ The following `CultureSelector` component shows how to call the `Set` method of
...
+
+    protected override void OnInitialized()
+    {
+        selectedCulture = CultureInfo.CurrentCulture;
+    }
+
+    private async Task HandleSelectedCultureChanged(ChangeEventArgs args)
+    {
+        selectedCulture = CultureInfo.GetCultureInfo((string)args.Value!);
+        if (CultureInfo.CurrentCulture != selectedCulture)
```
Comment (guardrex) [inline aspnetcore/blazor/globalization-localization.md:826]:
```suggestion

        if (CultureInfo.CurrentCulture != selectedCulture)
```

File: aspnetcore/blazor/globalization-localization.md [hunk 2]
```diff
@@ -483,25 +485,78 @@ The following `CultureSelector` component shows how to perform the following act
...
```
Comment (guardrex) [inline aspnetcore/blazor/globalization-localization.md:546]:
```suggestion

        if (CultureInfo.CurrentCulture != selectedCulture)
```

**Review by guardrex — APPROVED** [review]
Super-tiny-👶-NIT ... I'm going to place a blank line prior to the `if` lines. That's just to match our convention for `if`-blocks following other lines of code. **UPDATE**: *Done!* 👍 

Fantastic set of updates ... ***again!*** Is there any such thing as ***Blazor MVP of the Year***? There should be! I'll mention this one to Dan and Artak this Friday.

WRT the culture storage approach (local storage or loc cookie) for BWA apps, Mackinnon will be along in July/August to break the tie. He's going to decide which approach the article will take. Of course, I bet on Hisham because I'm I silly green hack-o-saurus 🦖, and Hisham is like you ... a *REAL* programmer! 😆

**Comment by hakenr** [comment]
> Fantastic set of updates ... again! Is there any such thing as Blazor MVP of the Year? There should be! I'll mention this one to Dan and Artak this Friday.

While I appreciate the praise as much as anyone, there's no need to overdo it.  I'm just doing what I enjoy. ;-)

**Pull request merged**